asserts/gpgkeypairmgr.go

```diff
@@ -298,7 +298,25 @@ func (gkm *GPGKeypairManager) Export(name string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return EncodePublicKey(keyInfo.pubKey)
+	return EncodePublicKey(keyInfo.privKey.PublicKey())
+}
+
+// SignWithoutAuthority assembles an assertion without a set authority with the provided information and signs it with the named key.
+func (gkm *GPGKeypairManager) SignWithoutAuthority(assertType *AssertionType, makeHeaders func(pubKey PublicKey) (map[string]interface{}, error), makeBody func(pubKey PublicKey) ([]byte, error), name string) (Assertion, error) {
+	keyInfo, err := gkm.findByName(name)
+	if err != nil {
+		return nil, err
+	}
+	pubKey := keyInfo.privKey.PublicKey()
+	headers, err := makeHeaders(pubKey)
```

Comment:
This method's signature feels a bit strange. Seems to invert the control of information just to avoid taking the key out. Why isn't the call site doing the generation of headers and body locally, and calling the actual SignWithAuthority locally as well?  All it needs is the private key for that. We have a method for taking the key by ID, and we can also have `GetWithName` to do the same using a name, which is effectively what we do here.